### PR TITLE
Added limit to avoid stackoverflows when calculating visible bounds

### PR DESCRIPTION
**Pull request by tri-fraga**
Described in [Appium #12892](https://github.com/appium/appium/issues/12892)

### CODE REVIEW

File: app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java [hunk 1]
```diff
@@ -36,6 +36,8 @@
  * This class contains static helper methods to work with {@link AccessibilityNodeInfo}
  */
 public class AccessibilityNodeInfoHelpers {
+	
+	private final static int MAX_DEPTH = 70;
```
Comment (mykola-mokhnach) [inline app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java]: please respect the indentation

File: app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java [hunk 2]
```diff
@@ -75,14 +77,24 @@ public static String getText(@Nullable AccessibilityNodeInfo nodeInfo, boolean r
         }
         return charSequenceToString(nodeInfo.getText(), replaceNull);
     }
-
-    /**
+	
+	    /**
```
Comment (mykola-mokhnach) [inline app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java]: same here

File: app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java (hunk 1 above)
Comment (mykola-mokhnach) [reply app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java]: same below

File: app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java [hunk 3]
```diff
@@ -96,16 +108,18 @@ public static Rect getVisibleBounds(@Nullable AccessibilityNodeInfo node) {
...
-        // Find the visible bounds of our first scrollable ancestor
-        for (AccessibilityNodeInfo ancestor = node.getParent(); ancestor != null; ancestor = ancestor.getParent()) {
+		// Find the visible bounds of our first scrollable ancestor 
+		for (AccessibilityNodeInfo ancestor = node.getParent(); ancestor != null && depth < MAX_DEPTH; ancestor = ancestor.getParent()) {
             // If this ancestor is scrollable
             if (ancestor.isScrollable()) {
                 // Trim any portion of the bounds that are hidden by the non-visible portion of our
                 // ancestor
-                Rect ancestorRect = getVisibleBounds(ancestor);
+                Rect ancestorRect = getVisibleBounds(ancestor, depth);
```
Comment (mykola-mokhnach) [inline app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java]: depth + 1

File: app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java [hunk 4]
```diff
@@ -96,16 +108,18 @@ public static Rect getVisibleBounds(@Nullable AccessibilityNodeInfo node) {
...
                 ret.intersect(ancestorRect);
                 break;
             }
+			
+			depth++;
```
Comment (mykola-mokhnach) [inline app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java]: this increment is not needed if the above fix is applied

File: app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java [hunk 5]
```diff
@@ -36,6 +36,8 @@
  * This class contains static helper methods to work with {@link AccessibilityNodeInfo}
  */
 public class AccessibilityNodeInfoHelpers {
+
```
Comment (mykola-mokhnach) [inline app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java]: please add a comment here with a link to the original UIA2 bug, which enforces us to have this workaround

File: app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java (hunk 4 above)
Comment (mykola-mokhnach) [inline app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java]: depth should be incremented before the condition

File: app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java (hunk 1 above)
Comment (tri-fraga) [reply app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java]: Done, should setup my IDE properly

File: app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java (hunk 4 above)
Comment (tri-fraga) [reply app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java]: Done

**Pull request merged**